### PR TITLE
Deploy with morph, in buildkite, with vault and terraform

### PR DESCRIPTION
**Pull request by grahamc**
what a mess of a PR, there is a lot of goop in the repo to clean up.

### CODE REVIEW

File: terraform-old/core.tf [hunk 1]
```diff
@@ -31,7 +31,7 @@ resource "packet_device" "core" {
   count            = 1 # nixops code doesn't support > 1 right now
   hostname         = "core-${count.index}.ewr1.nix.ci"
   plan             = "baremetal_0"
-  facility         = "ewr1"
+  facilities       = [ "ewr1" ]
```
Comment (cole-h) [inline terraform-old/core.tf:34]: Should we just `rm -r terraform-old`? It's still present in git history if we ever need to refer to it again.

File: ci/terraform-import.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
+env:
+  GIT_AUTHOR_NAME: Buildkite
+  GIT_AUTHOR_EMAIL: ci@buildkite
+  GIT_COMMITTER_NAME: Buildkite
+  GIT_COMMITTER_EMAIL: ci@buildkite
+  NIX_PATH: nixpkgs=channel:nixos-unstable-small
+  NIXPKGS_ALLOW_UNFREE: 1
```
Comment (cole-h) [inline ci/terraform-import.yml:7]: /me wonders what this is for

File: enter-env.sh [hunk 3]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -I nixpkgs=channel:nixos-unstable-small -p bashInteractive vault awscli jq openssh
+
+set +x # don't leak secrets!
+set -eu
+
+scratch=$(mktemp -d -t tmp.XXXXXXXXXX)
+function finish {
+  git remote rm vaultpush || true
```
Comment (cole-h) [inline enter-env.sh:9]: Maybe?

```suggestion
  git remote rm vaultpush 2>/dev/null || true
```

File: enter-env.sh [hunk 4]
```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -I nixpkgs=channel:nixos-unstable-small -p bashInteractive vault awscli jq openssh
```
Comment (cole-h) [inline enter-env.sh:2]: I think we should only need either `-I nixpkgs=` or `NIX_PATH`, right?

File: terraform/base/packet.tf [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+        },
+        {
+          dataset ="npool/var"
+          point ="/var"
+        }
+      ]
+    }
+  })
+
+  # ipxe_script_url     = "http://images.platformequinix.net/nixos/installer-pre2/x86/netboot.ipxe"
```
Comment (cole-h) [inline terraform/base/packet.tf:148]: 🧹

File: terraform/base/packet.tf [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+  operating_system = "custom_ipxe"
+  plan             = "m1.xlarge.x86"
+  facilities       = ["ewr1"]
+
+  user_data = <<USERDATA
+#!nix
+${var.bootstrap_expr}
+USERDATA
+
+#  custom_data = jsonencode({
```
Comment (cole-h) [inline terraform/base/packet.tf:172]: 🧹

File: terraform/base/packet.tf [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+#        },
+#        {
+#          dataset ="npool/var"
+#          point ="/var"
+#        }
+#      ]
+#    }
+#  })
+
+  # ipxe_script_url     = "http://images.platformequinix.net/nixos/installer-pre2/x86/netboot.ipxe"
```
Comment (cole-h) [inline terraform/base/packet.tf:256]: 🧹

File: terraform/shell.nix [hunk 8]
```diff
@@ -0,0 +1,84 @@
+let
+  sources = import ./nix/sources.nix;
+  overlay = _: pkgs: {
```
Comment (cole-h) [inline terraform/shell.nix:3]: 🧹

**Review by cole-h — COMMENTED** [review]
Woohoo!

**Comment by cole-h** [comment]
Also, dunno if this is on me (e.g. my Firefox configuration) or not, but something seems to be up with Grafana. Any graph I try to make just gives me `a is undefined`, and I can't find any endpoint that gives me access to the `ofborg_queue_evaluator_in_progress` -- and other -- metrics. Maybe related to the rabbitmq host change? Dunno. Weird stuff.

File: make-targets.sh [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+      ./machines/$name.system.nix
+    ];
+  };
+EOF
+)
+
+cat <<EOF > "$scratch/default.nix"
+{
+  network = {
+    pkgs = import <nixpkgs> {};
```
[the remark this answers is not in the document]
Comment (grahamc) [reply make-targets.sh:43]: This is why we need that ALLOW_UNFREE bit, and if we simply set the config here, we wouldn't need the environment variable. Agreed on :(. Maybe you can help get away from this?

File: make-targets.sh [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        name="$(jq -r .key <<<"$machine")"
+        ip=$(jq -r .value.ip <<<"$machine")
+        jq -r .value.expression <<<"$machine"
+        jq -r .value.expression <<<"$machine" > "$scratch/machines/${name}.expr.nix"
+        if ssh \
+          -o StrictHostKeyChecking=no \
+          -o UserKnownHostsFile="$scratch/known_hosts" \
+          -o BatchMode=yes \
+          -o IdentitiesOnly=yes \
+          -i ./deploy.key \
```
[the remark this answers is not in the document]
Comment (grahamc) [reply make-targets.sh:63]: :+1:

```suggestion
          -i "$SSH_IDENTITY_FILE" \
```

File: nixops/modules/rabbitmq/default.nix [hunk 11]
```diff
@@ -114,6 +54,12 @@ in {
       };
       environment = {
         PUBLISH_PORT = "9419";
+        RABBIT_URL = "https://${cfg.domain}";
+        SKIPVERIFY = "true"; # yeah, sue me. the current version in nixpkgs
```
[the remark this answers is not in the document]
Comment (grahamc) [reply nixops/modules/rabbitmq/default.nix:58]: Maybe you can help resolve the comment? :)

File: nixops/modules/site.nix [hunk 12]
```diff
@@ -25,7 +26,7 @@ in {
     ];
 
     services.ofborg.webhook.domain = "webhook.nix.ci";
-    services.ofborg.rabbitmq.domain = "events.nix.ci";
+    services.ofborg.rabbitmq.domain = "devoted-teal-duck.rmq.cloudamqp.com";
```
[the remark this answers is not in the document]
Comment (grahamc) [reply nixops/modules/site.nix:29]: :eyes:

File: terraform-old/core.tf (hunk 1 above)
Comment (grahamc) [reply terraform-old/core.tf]: :heavy_check_mark:

File: terraform/shell.nix (hunk 8 above)
Comment (grahamc) [reply terraform/shell.nix:3]: Eh, I kind of want to keep it in case there are further additions.

File: ci/terraform-import.yml (hunk 2 above)
Comment (grahamc) [reply ci/terraform-import.yml:7]: For what it is worth, I'd rather totally avoid setting the NIX_PATH (environment variable) and this NIXPKGS_ALLOW_UNFREE variable anywhere.

File: nixops/modules/rabbitmq/default.nix [hunk 13]
```diff
@@ -114,6 +54,12 @@ in {
...
+        # fails on https://github.com/kbudde/rabbitmq_exporter/issues/149
+        # and while I'd like to upgrade it, it was a can of worms... and
+        # this username and password isn't very interesting, anyway.
+
         RABBIT_CAPABILITIE = "bert,no_sort";
```
Comment (cole-h) [inline nixops/modules/rabbitmq/default.nix:63]: Not in this diff, but, uh, should this be `RABBIT_CAPABILITIES`?

File: nixops/modules/rabbitmq/default.nix (hunk 11 above)
[the remark this answers is not in the document]
Comment (cole-h) [reply nixops/modules/rabbitmq/default.nix:58]: https://github.com/NixOS/nixpkgs/pull/109495

File: nixops/modules/rabbitmq/default.nix (hunk 13 above)
Comment (grahamc) [reply nixops/modules/rabbitmq/default.nix:63]: lol, I wonder what the story is here. Let's delete it.

```suggestion
```

**Pull request merged**